addons/survey/controllers/main.py

```diff
@@ -291,9 +282,30 @@ def survey_display_page(self, survey_token, answer_token, **post):
             if is_last:
                 data.update({'last': True})
 
-            return request.render('survey.survey_page_main', data)
+            return data
         else:
-            return request.render("survey.survey_403_page", {'survey': survey_sudo})
+            return {'not_found': True}
```

Comment:
When does this happen exactly?

Controllers call `_check_validity` with `ensure_token` set to `True` when they require an answer.